packages/main/src/RangeSlider.js

```diff
@@ -0,0 +1,382 @@
+ *
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.RangeSlider
+ * @extends sap.ui.webcomponents.base.UI5Element
+ * @tagname ui5-range-slider
+ * @since 1.0.0-rc.11
+ * @appenddocs SliderBase
+ * @public
+ */
+class RangeSlider extends SliderBase {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get template() {
+		return RangeSliderTemplate;
+	}
+
+	constructor() {
+		super();
+		this._stateStorage.startValue = null;
+		this._stateStorage.endValue = null;
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+	}
+
+	get tooltipStartValue() {
+		const stepPrecision = this.constructor._getDecimalPrecisionOfNumber(this._effectiveStep);
+		return this.startValue.toFixed(stepPrecision);
+	}
+
+	get tooltipEndValue() {
+		const stepPrecision = this.constructor._getDecimalPrecisionOfNumber(this._effectiveStep);
+		return this.endValue.toFixed(stepPrecision);
+	}
+
+	/**
+	 * Check if the previously saved state is outdated. That would mean
+	 * either it is the initial rendering or that a property has been changed
+	 * programatically - because the previous state is always updated in
+	 * the interaction handlers.
+	 *
+	 * Normalize current properties, update the previously stored state.
+	 * Update the visual UI representation of the Slider.
+	 *
+	 */
+	onBeforeRendering() {
+		if (!this.isCurrentStateOutdated()) {
+			return;
+		}
+
+		this.notResized = true;
+		this.syncUIAndState("startValue", "endValue");
+		this._updateHandlesAndRange(null);
+	}
+
+	/**
+	 * Called when the user starts interacting with the slider
+	 *
+	 * @private
+	 */
+	_onmousedown(event) {
+		// If step is 0 no interaction is available because there is no constant
+		// (equal for all user environments) quantitative representation of the value
+		if (this.disabled || this._effectiveStep === 0) {
+			return;
+		}
+
+		const oldEndValue = this.endValue;
+		const oldStartValue = this.startValue;
+		const newValue = this.handleDownBase(event, this._effectiveMin, this._effectiveMax);
+
+		// Check if the new value is in the current select range of values
+		this._isNewValueInCurrentRange = newValue > oldStartValue && newValue < oldEndValue;
+		// Save the initial press point coordinates (position).
+		this._initialPageXPosition = this.constructor.getPageXValueFromEvent(event);
+		// Determine value to be modified depending on the the handle clossest to the press point
+		this._setCurrentValueType(this._initialPageXPosition, newValue);
+
+		// Use the progress bar to save the initial coordinates of the start-handle when the interaction begins.
+		// We will use it as a reference to calculate a moving offset if the whole range selection is dragged.
+		this._initialStartHandlePageX = this.directionStart === "left" ? this.shadowRoot.querySelector(".ui5-slider-progress").getBoundingClientRect().left : this.shadowRoot.querySelector(".ui5-slider-progress").getBoundingClientRect().right;
```

Comment:
Code duplication, extract:
this.shadowRoot.querySelector(".ui5-slider-progress").getBoundingClientRect()
to a new variable before this line